### ISSUE TITLE
AdaptedAttention doesn't work with model loaded in 8bit or 4bit

### ISSUE THREAD

**Issue by nuance1979** [huggingface/peft#600]
### System Info

Python 3.8.16
peft 0.4.0.dev0
transformers 4.29.2
accelerate 0.20.3

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder
- [X] My own task or dataset (give details below)

### Reproduction

```python
import transformers
from peft import AdaptionPromptConfig, get_peft_model, prepare_model_for_kbit_training

model = transformers.LlamaForCausalLM.from_pretrained(
        "decapoda-research/llama-7b-hf",
        load_in_8bit=True,
        torch_dtype=torch.float16,
        device_map="auto",
)

config = AdaptionPromptConfig(
    adapter_len=10,
    adapter_layers=30,
    task_type="CAUSAL_LM",
)
model = get_peft_model(model, config)
```

### Expected behavior

The above code will throw at the this line, complaining that `Char` type tensor can't do `normal_()`:

https://github.com/huggingface/peft/blob/08cb3dde577747f6ca6638c884fd66fd16cf2e9d/src/peft/tuners/adaption_prompt.py#L297-L299

This is because, unlike `LoraModel`, `AdaptationPromptModel` is not aware of the possibility that the base model can be load in 8bit so that `model.q_proj.weight.dtype` is now a `torch.int8`. I manually change it to `torch.float32` and it works fine.

In summary, `AdaptationPromptModel` should check `loaded_in_8bit = getattr(self.model, "is_loaded_in_8bit", False)` then set the correct float type. Somewhere a `compute_dtype` or similar should be passed in.

**Comment by younesbelkada** [comment]
Hi @nuance1979 
Thanks for the PR and the clean reproducer. https://github.com/huggingface/peft/pull/604 should add the bitsandbytes (4bit & 8bit) support of AdaptionPrompt
